exec/java-exec/src/test/java/org/apache/drill/exec/server/TestSpnegoAuthentication.java

```diff
@@ -0,0 +1,597 @@
+        .withValue(ExecConstants.HTTP_AUTHENTICATION_MECHANISMS,
+            ConfigValueFactory.fromIterable(Lists.newArrayList("spnego")))
+        .withValue(ExecConstants.USER_AUTHENTICATION_ENABLED,
+            ConfigValueFactory.fromAnyRef(true))
+        .withValue(ExecConstants.HTTP_SPNEGO_PRINCIPAL,
+            ConfigValueFactory.fromAnyRef(spnegoHelper.SERVER_PRINCIPAL))
+        .withValue(ExecConstants.HTTP_SPNEGO_KEYTAB,
+            ConfigValueFactory.fromAnyRef(spnegoHelper.serverKeytab.toString())),
+        false);
+
+    final ScanResult scanResult = ClassPathScanner.fromPrescan(newConfig);
+    final AuthenticatorProviderImpl authenticatorProvider = Mockito.mock(AuthenticatorProviderImpl.class);
+    Mockito.when(authenticatorProvider.containsFactory(PlainFactory.SIMPLE_NAME)).thenReturn(false);
+
+    final DrillbitContext context = Mockito.mock(DrillbitContext.class);
+    Mockito.when(context.getClasspathScan()).thenReturn(scanResult);
+    Mockito.when(context.getConfig()).thenReturn(newConfig);
+    Mockito.when(context.getAuthProvider()).thenReturn(authenticatorProvider);
+
+    final DrillHttpSecurityHandlerProvider securityProvider = new DrillHttpSecurityHandlerProvider(newConfig, context);
+
+    assertTrue(!securityProvider.isFormEnabled());
+    assertTrue(securityProvider.isSpnegoEnabled());
+  }
+
+  /**
+   * Validate when none of the security mechanism is specified in the
+   * {@link ExecConstants#HTTP_AUTHENTICATION_MECHANISMS}, FORM security handler is still configured correctly when
+   * authentication is enabled along with PAM authenticator module.
+   * @throws Exception
+   */
+  @Test
+  public void testConfigBackwardCompatibility() throws Exception {
+
+    final DrillConfig newConfig = new DrillConfig(DrillConfig.create()
+        .withValue(ExecConstants.USER_AUTHENTICATION_ENABLED,
+            ConfigValueFactory.fromAnyRef(true)),
+        false);
+
+    final ScanResult scanResult = ClassPathScanner.fromPrescan(newConfig);
+    final AuthenticatorProviderImpl authenticatorProvider = Mockito.mock(AuthenticatorProviderImpl.class);
+    Mockito.when(authenticatorProvider.containsFactory(PlainFactory.SIMPLE_NAME)).thenReturn(true);
+
+    final DrillbitContext context = Mockito.mock(DrillbitContext.class);
+    Mockito.when(context.getClasspathScan()).thenReturn(scanResult);
+    Mockito.when(context.getConfig()).thenReturn(newConfig);
+    Mockito.when(context.getAuthProvider()).thenReturn(authenticatorProvider);
+
+    final DrillHttpSecurityHandlerProvider securityProvider = new DrillHttpSecurityHandlerProvider(newConfig, context);
+
+    assertTrue(securityProvider.isFormEnabled());
+    assertTrue(!securityProvider.isSpnegoEnabled());
+  }
+
+  /**
+   * Validate behavior of {@link SpnegoUtil} class when provided with different configuration's for SPNEGO via
+   * DrillConfig
+   * @throws Exception
+   */
+  @Test
+  public void testSpnegoUtil() throws Exception {
```

Comment:
Test for spnego util can be factored out in different class plus test can be split into four.